src/Compilers/Core/Portable/InternalUtilities/MultiDictionary.cs

```diff
@@ -95,6 +95,10 @@ public bool MoveNext()
             // Stores either a single V or an ImmutableHashSet<V>
             private readonly object _value;
 
+            private readonly IEqualityComparer<V> _equalityComparer;
+
+            private IEqualityComparer<V> EqualityComparer => _equalityComparer ?? ImmutableHashSet<V>.Empty.KeyComparer;
```

Comment:
>_equalityComparer ?? ImmutableHashSet<V>.Empty.KeyComparer [](start = 61, length = 58)

Consider storing this non-null value in the field.

After that, do we need the property?

Reply:
We can't, because its a struct and not every call goes through the constructor, or provides a comparer.

---
In reply to: [305441552](https://github.com/dotnet/roslyn/pull/37247#discussion_r305441552) [](ancestors = 305441552)